api/src/services/fields.ts

```diff
@@ -944,4 +962,60 @@ export class FieldsService {
 			column.alter();
 		}
 	}
+
+	private async trackFieldSchemaChange(
+		trx: Knex.Transaction,
+		collection: string,
+		fieldName: string,
+		action: 'create' | 'update',
+		schemaData: any,
+	): Promise<void> {
+		if (!this.accountability || this.schema.collections['directus_fields']?.accountability === null) {
+			return;
+		}
+
+		const fieldRecord = await trx('directus_fields').select('id').where({ collection, field: fieldName }).first();
+
+		if (!fieldRecord) {
+			throw new Error(`Field ${collection}.${fieldName} not found`);
+		}
+
+		const { ActivityService } = await import('./activity.js');
+
+		const activityService = new ActivityService({
+			knex: trx,
+			schema: this.schema,
+		});
+
+		const activity = await activityService.createOne({
+			action,
+			user: this.accountability.user,
+			collection: 'directus_fields',
+			ip: this.accountability.ip,
+			user_agent: this.accountability.userAgent,
+			origin: this.accountability.origin,
+			item: fieldRecord.id,
+		});
+
+		if (this.schema.collections['directus_fields']?.accountability === 'all') {
```

Comment:
This check should be unnecessary as our system collections can't change their accountability. It's the responsibility of the core to determine what's considered "necessary" in this context.